website/docs/en/api/loader-api/context.mdx

```diff
@@ -0,0 +1,114 @@
+import WebpackLicense from '@components/webpack-license';
+
+<WebpackLicense from="https://webpack.js.org/api/loaders/#the-loader-context" />
+
+# The Loader Context
+
+The loader context represents the properties that are available inside of a loader assigned to the `this` property.
+
+## `this.addContextDependency(directory: string)`
+
+Add the directory as a dependency for the loader results so that any changes to the files in the directory can be listened to.
+
+## `this.addDependency(file: string)`
+
+Add a file as a dependency on the loader results so that any changes to them can be listened to. For example, `sass-loader`, `less-loader` use this trick to recompile when the imported style files change.
+
+## `this.dependency(file: string)`
+
+Alias of `this.addDependency(file: string)`.
+
+## `this.addMissingDependency(file: string)`
+
+Add a non-existent file as a dependency on the loader results to make them listenable.
+
+## `this.async()`
+
+Tells Rspack that this loader will be called asynchronously. Returns `this.callback`.
+
+## `this.cacheable(flag: boolean = true)`
+
+By default, the processing results of the loader are marked as cacheable. Calling this method and passing `false` turns off the loader's ability to cache processing results.
+
+## `this.callback(err: Error | null, content: string | Buffer, sourceMap?: SourceMap, meta?: any)`
+
+A function that can be called synchronously or asynchronously in order to return multiple results. The expected arguments are:
+
+1. The first parameter must be `Error` or `null`, which marks the current module as a compilation failure.
+2. The second argument is a `string` or `Buffer`, which indicates the contents of the file after the module has been processed by the loader.
+3. The third parameter is a source map that can be processed by the loader.
+4. The fourth parameter is ignored by Rspack and can be anything (e.g. some metadata).
+
+## `this.clearDependencies()`
+
+Removes all dependencies of the loader result.
+
+## `this.context`
+
+The directory where the current module is located.
+
+## `this.data`
+
+A data object shared between the pitch and the normal phase.
+
+## `this.emitError(err: Error)`
+
+Emit an error. Unlike `throw` and `this.callback(err)` in the loader, it does not mark the current module as a compilation failure, it just adds an error to Rspack's Compilation and displays it on the command line at the end of this compilation.
+
+## `this.emitWarning(warning: Error)`
+
+Emit a warning.
+
+## `this.emitFile(name: string, content: Buffer | string, sourceMap: SourceMap)`
+
+Emit a file.
+
+## `this.getOptions(schema)`
+
+Extracts the given loader option, accepting an optional JSON schema as an argument.
+
+## `this.getResolve(options: ResolveOptions): resolve`
+
+Create a resolver like `this.resolve`.
+
+## `this.resolve(context: string, request: string, callback: (err: Error | null, result: string) => void)`
```

Comment:
The title is too long, can we only show the function name in the title? like this:

````markdown
## this.resolve

- Type:

```ts
function resolve(
  context: string,
  request: string,
  callback: (err: Error | null, result: string) => void,
): void
```
````